src/accelerate/checkpointing.py

```diff
@@ -174,31 +179,39 @@ def load_accelerator_state(
         map_location = "cpu"
     elif map_location == "on_device":
         map_location = PartialState().device
+
+    input_dir = Path(input_dir)
     # Model states
     for i, model in enumerate(models):
-        weights_name = f"{MODEL_NAME}.bin" if i == 0 else f"{MODEL_NAME}_{i}.bin"
-        input_model_file = os.path.join(input_dir, weights_name)
-        models[i].load_state_dict(torch.load(input_model_file, map_location=map_location), **load_model_func_kwargs)
+        weights_name = f"{MODEL_NAME}.safetensors" if i == 0 else f"{MODEL_NAME}_{i}.safetensors"
```

Comment:
Just to be sure I understand correctly, here we are "guessing" that the checkpoint uses safetensors and if it doesn't, we assume it uses the PyTorch format.

Reply:
Correct!

Reply:
We'd still have to guess, but we can simplify the logic some :) (As during `load` we don't know which to use)